postgres/parser/parser/parse.go

```diff
@@ -231,6 +231,16 @@ func unaryNegation(e tree.Expr) tree.Expr {
 	return &tree.UnaryExpr{Operator: tree.UnaryMinus, Expr: e}
 }
 
+func absolute(e tree.Expr) tree.Expr {
```

Comment:
I don't think we want to actually do this. It's something that I'd need to test, but raw literals are treated differently than the output of functions, and the absolute operator is internally a function call.

```sql
SELECT oid, oprkind, oprleft, oprright, oprcode FROM pg_operator o WHERE o.oprname = '@' ORDER BY oprcode::varchar;
```

So doing the simplification in the parser may result in differing output later in the pipeline. For now, it's safer to always return the unary expression, we can do the simplification later if what I mentioned above proves untrue.